support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -183,10 +201,45 @@ func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		return makeID(r.c.TrustDomain, r.getIdentityTemplate(pod))
+	}
+
+	// the controller has not been configured with a specific identity format.
 	// create an entry based on the service account.
 	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
 }
 
 func (r *PodReconciler) podParentID(nodeName string) string {
 	return makeID(r.c.TrustDomain, "k8s-workload-registrar/%s/node/%s", r.c.Cluster, nodeName)
 }
+
+func (r *PodReconciler) getIdentityTemplate(pod *corev1.Pod) string {
+	tpl := r.c.IdentityTemplate
+	tmpl, err := template.New("IdentityTemplate").Parse(tpl)
+	if err != nil {
+		r.c.Log.WithError(err).Errorf("Error parsing the template (%v)", tpl)
```

Comment:
looks like to me that if template can not be created it must fails here, since tmpl will be nil and panic on line 240

Reply:
what to you think about `Errorf("Error parsing the template %q", tpl`?

Reply:
Sounds good. Fixed!